### ISSUE TITLE
ArrayExpression & ObjectExpression indent configs do not work?

### ISSUE THREAD

**Issue by JakeSidSmith** [eslint/eslint#8437]
Not sure if this is actually a bug, or I'm just doing something wrong, but I'm pretty sure I've followed the docs accurately.

**Tell us about your environment**

* **ESLint Version:**

`3.10.0` / `3.19.0`

* **Node Version:**

`6.10.2`

* **npm Version:**

`3.10.10`

**What parser (default, Babel-ESLint, etc.) are you using?**

`default`

**Please show your full configuration:**

```json
"indent": [
  2,
  2,
  {
    "SwitchCase": 1,
    "outerIIFEBody": 1,
    "MemberExpression": 1,
    "ArrayExpression": 1,
    "ObjectExpression": 1
  }
]
```

**What did you do? Please include the actual source code causing the issue.**

`n/a`

**What did you expect to happen?**

I expected the config to be valid.

**What actually happened? Please include the actual, raw output from ESLint.**

```shell
Error: /project/.eslintrc.json:
	Configuration for rule "indent" is invalid:
	Value "data["1"].ArrayExpression" has additional properties.
	Value "data["1"].ObjectExpression" has additional properties.

```

**Comment by gyandeeps** [comment]
I can't reproduce this. Can you makesure you dont have indent configured incorrectly in some nested .eslintrc file.

**Comment by JakeSidSmith** [comment]
I've tried running with literally only the config above, and still get errors. 😞

**Comment by gyandeeps** [comment]
Can you try to run with `--debug` flag to makesure there no other config file getting picked up. Also can u share the output of this command on a simplified case.

**Comment by JakeSidSmith** [comment]
```
> eslint --debug -c .eslintrc.json test-pass.js

  eslint:cli Running on files +0ms
  eslint:config Using command line config .eslintrc.json +23ms
  eslint:config-file Loading JSON config file: /project/.eslintrc.json +3ms
/project/.eslintrc.json:
	Configuration for rule "indent" is invalid:
	Value "data["1"].ArrayExpression" has additional properties.
	Value "data["1"].ObjectExpression" has additional properties.

Error: /project/.eslintrc.json:
	Configuration for rule "indent" is invalid:
	Value "data["1"].ArrayExpression" has additional properties.
	Value "data["1"].ObjectExpression" has additional properties.

    at validateRuleOptions (/project/node_modules/eslint/lib/config/config-validator.js:116:15)
    at /project/node_modules/eslint/lib/config/config-validator.js:163:13
    at Array.forEach (native)
    at Object.validate (/project/node_modules/eslint/lib/config/config-validator.js:162:35)
    at Object.load (/project/node_modules/eslint/lib/config/config-file.js:523:19)
    at loadConfig (/project/node_modules/eslint/lib/config.js:64:33)
    at new Config (/project/node_modules/eslint/lib/config.js:235:38)
    at CLIEngine.executeOnFiles (/project/node_modules/eslint/lib/cli-engine.js:594:28)
    at Object.execute (/project/node_modules/eslint/lib/cli.js:181:101)
    at Object.<anonymous> (project/node_modules/eslint/bin/eslint.js:76:28)
```

**Comment by not-an-aardvark** [comment]
> * **ESLint Version:**
> `3.10.0` / `3.19.0`

Could you clarify this? The `ArrayExpression` and `ObjectExpression` options were added in 3.12.0, so the version of eslint that you're using is important here.

**Comment by JakeSidSmith** [comment]
Ah, I tried initially in `3.10` and it didn't work, so I upgraded, cleared my modules & tried again, and it still didn't work. I'll just confirm that...

**Comment by JakeSidSmith** [comment]
Okay, my bad. Seems I'd messed up my dependencies.

Had accidentally removed my dev dependency and one of my dependencies was installing `3.2`.

Resolved. :) Sorry for wasting your time.
